ftw/http.py

```diff
@@ -334,13 +335,13 @@ def build_request(self):
         headers = ''
         if self.request_object.headers != {}:
             for hname, hvalue in self.request_object.headers.iteritems():
-                headers += str(hname) + ': ' + str(hvalue) + str(self.CRLF)
+                headers += unicode(hname) + ': ' + unicode(hvalue) + str(self.CRLF)
         request = string.replace(request, '#headers#', headers)
 
         # If we have data append it
         if self.request_object.data != '':
             data = str(self.request_object.data)
-            request = string.replace(request, '#data#', data)
+            request = string.replace(request, '#data#', data + self.CRLF)
```

Comment:
HTTP requires a trailing /r/n which was manually added before but it's nicer if it works like magic.